MSBuild integration doesn't work for wxWidgets package

### Synopsis

wxWidgets package installs one of its headers in `$VcpkgRoot\lib\mswu\wx\setup.h` where it's not found because the default headers path only contains `$VcpkgRoot\include`.

### How to reproduce

Perform the following steps (tested under Windows 7 x64 in 32 bit command line prompt, but shouldn't be OS-dependent):

1. Install vcpkg, by cloning it from the repository and running the bootstrap script.
1. Run `.\vcpkg.exe integrate install`.
1. Run `.\vcpkg.exe install wxwidgets`.
1. Create a new empty Win32 desktop project in MSVS 2017.
1. Create a new C++ source file containing `#include <wx/wx.h>` line.
1. Try to compile it, resulting in an error message similar to the following: `...\vcpkg\installed\x86-windows\include\wx\platform.h(148): fatal error C1083: Cannot open include file: 'wx/setup.h': No such file or directory`

### Solution

I'm not sure what would be the best way to solve this. Adding `$VcpkgRoot\lib\mswu` would fix this for the release builds, but in Debug the header is under `$VcpkgRoot\debug\lib\mswud`, so the include paths would need to differ depending on the configuration.

Alternatively, the file could be installed as `$VcpkgRoot\include\wx\setup.h` instead of its usual location because it doesn't actually depend on the configuration (in principle, it could, but with vcpkg build only `wxINSTALL_PREFIX` differs and this is not really important). This is probably the simplest thing to do.

--- a/ports/wxwidgets/CONTROL
+++ b/ports/wxwidgets/CONTROL
@@ -1,4 +1,4 @@
 Source: wxwidgets
-Version: 3.1.1-1
+Version: 3.1.1-2
 Description: wxWidgets is a widget toolkit and tools library for creating graphical user interfaces (GUIs) for cross-platform applications.
 Build-Depends: zlib, libpng, tiff, expat

--- a/ports/wxwidgets/CONTROL
+++ b/ports/wxwidgets/CONTROL
@@ -1,4 +1,4 @@
 Source: wxwidgets
-Version: 3.1.1
+Version: 3.1.1-1
 Description: wxWidgets is a widget toolkit and tools library for creating graphical user interfaces (GUIs) for cross-platform applications.
 Build-Depends: zlib, libpng, tiff, expat

--- a/ports/wxwidgets/portfile.cmake
+++ b/ports/wxwidgets/portfile.cmake
@@ -48,4 +48,8 @@ endif()
 # Handle copyright
 file(COPY ${SOURCE_PATH}/docs/licence.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/wxwidgets)
 file(RENAME ${CURRENT_PACKAGES_DIR}/share/wxwidgets/licence.txt ${CURRENT_PACKAGES_DIR}/share/wxwidgets/copyright)
+
+file(RENAME ${CURRENT_PACKAGES_DIR}/lib/mswu/wx/setup.h ${CURRENT_PACKAGES_DIR}/include/wx/setup.h)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib/mswu)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib/mswud)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)

--- a/ports/wxwidgets/portfile.cmake
+++ b/ports/wxwidgets/portfile.cmake
@@ -52,4 +52,5 @@ file(RENAME ${CURRENT_PACKAGES_DIR}/share/wxwidgets/licence.txt ${CURRENT_PACKAG
 file(RENAME ${CURRENT_PACKAGES_DIR}/lib/mswu/wx/setup.h ${CURRENT_PACKAGES_DIR}/include/wx/setup.h)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib/mswu)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib/mswud)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/include/msvc)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)

--- a/ports/wxwidgets/portfile.cmake
+++ b/ports/wxwidgets/portfile.cmake
@@ -5,11 +5,7 @@ vcpkg_from_github(
     REF v3.1.1
     SHA512 f6d8974e2f48bae7e96a8938df3ad5efc403036c1dcbe2b48edd276ee7923802ba3e95e3f3bd9db17985e427b8e4f78950df0cbba83ae99d508ed04633816c95
     HEAD_REF master
-)
-
-vcpkg_apply_patches(
-    SOURCE_PATH ${SOURCE_PATH}
-    PATCHES "${CMAKE_CURRENT_LIST_DIR}/disable-platform-lib-dir.patch"
+    PATCHES disable-platform-lib-dir.patch
 )
 
 vcpkg_configure_cmake(
@@ -49,7 +45,9 @@ endif()
 file(COPY ${SOURCE_PATH}/docs/licence.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/wxwidgets)
 file(RENAME ${CURRENT_PACKAGES_DIR}/share/wxwidgets/licence.txt ${CURRENT_PACKAGES_DIR}/share/wxwidgets/copyright)
 
-file(RENAME ${CURRENT_PACKAGES_DIR}/lib/mswu/wx/setup.h ${CURRENT_PACKAGES_DIR}/include/wx/setup.h)
+if(EXISTS ${CURRENT_PACKAGES_DIR}/lib/mswu/wx/setup.h)
+    file(RENAME ${CURRENT_PACKAGES_DIR}/lib/mswu/wx/setup.h ${CURRENT_PACKAGES_DIR}/include/wx/setup.h)
+endif()
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib/mswu)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib/mswud)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/include/msvc)

